components/authz-service/server/v2/policy.go

```diff
@@ -536,18 +536,18 @@ func (s *policyServer) MigrateToV2(ctx context.Context,
 		return nil, status.Errorf(codes.Internal, "retrieve default policies: %s", err.Error())
 	}
 
-	for _, pol := range defaultPolicies {
-		if _, err := s.store.CreatePolicy(ctx, &pol); err != nil {
-			return nil, status.Errorf(codes.Internal, "reset to default policies: %s", err.Error())
-		}
-	}
-
 	for _, role := range storage.DefaultRoles() {
 		if _, err := s.store.CreateRole(ctx, &role); err != nil {
 			return nil, status.Errorf(codes.Internal, "reset to default roles: %s", err.Error())
 		}
 	}
 
+	for _, pol := range defaultPolicies {
+		if _, err := s.store.CreatePolicy(ctx, &pol); err != nil {
+			return nil, status.Errorf(codes.Internal, "reset to default policies: %s", err.Error())
+		}
+	}
```

Comment:
Since roles have to actually exist now to be inserted into a policy statement, we need to create the roles first ;)

Reply:
Makes me wonder how it's OK to create the default projects _last_ (👇), but I suppose there are none, or they're not used.

Reply:
They are definitely used--that's where all-projects comes from. 😁  It might be the case that it is duplicating the effort of a migration that creates them first.

Reply:
well `~~all projects~~` is used in all the migrated statements 🤔